lib/validation.js

```diff
@@ -152,48 +166,60 @@ module.exports.process = function (validationModel, req, options) {
     var errors = options.errorsAsArray === false ? {} : [];
     var self = this;
 
-    function addError(error, key) {
```

Comment:
The `process` function was refactored and most of its work is now done in `validateObject` to allow recursive validation.